frontend/src/components/activePage/TableInstance/RowInstanceActions/RowInstanceActionsDropdown.tsx

```diff
@@ -9,64 +9,76 @@ import {
 } from '@ant-design/icons';
 import { ISSHInfo } from '../SSHModalContent/SSHModalContent';
 import { Instance } from '../../../../utils';
+import {
+  useApplyInstanceMutation,
+  useDeleteInstanceMutation,
+} from '../../../../generated-types';
+import { DropDownAction, setInstanceRunning } from '../../ActiveUtils';
 
 export interface IRowInstanceActionsDropdownProps {
   instance: Instance;
   fileManager?: boolean;
   ssh?: ISSHInfo;
   extended: boolean;
-  startInstance?: (idInstance: string, idTemplate: string) => void;
-  stopInstance?: (idInstance: string, idTemplate: string) => void;
-  destroyInstance: (instanceId: string, tenantNamespace: string) => void;
   setSshModal: React.Dispatch<SetStateAction<boolean>>;
 }
 
 const RowInstanceActionsDropdown: FC<IRowInstanceActionsDropdownProps> = ({
   ...props
 }) => {
-  const {
-    instance,
-    fileManager,
-    ssh,
-    extended,
-    startInstance,
-    stopInstance,
-    destroyInstance,
-    setSshModal,
-  } = props;
+  const { instance, fileManager, ssh, extended, setSshModal } = props;
+
+  const { status, persistent, url, name, tenantNamespace, running } = instance;
+
+  const [disabled, setDisabled] = useState(false);
+  const [deleteInstanceMutation] = useDeleteInstanceMutation();
+  const [applyInstanceMutation] = useApplyInstanceMutation();
 
-  const {
-    status,
-    persistent,
-    url,
-    idTemplate,
-    name,
-    tenantDisplayName,
-  } = instance;
+  const setInstanceStatus = async () => {
+    if (!disabled) {
+      setDisabled(true);
+      try {
+        const result = await setInstanceRunning(
+          running!,
+          instance,
+          applyInstanceMutation
+        );
+        if (result) setTimeout(() => setDisabled(false), 400);
+      } catch {
+        //do nothing at the moment
+      }
+    }
+  };
 
-  const dropdownHandler = (key: string) => {
+  const dropdownHandler = (key: DropDownAction) => {
     switch (key) {
-      case 'Start':
-        persistent && startInstance?.(name, idTemplate!);
-        break;
-      case 'Stop':
-        persistent && stopInstance?.(name, idTemplate!);
+      case DropDownAction.start || DropDownAction.stop:
+        persistent && setInstanceStatus();
         break;
-      case 'Connect':
+      case DropDownAction.connect:
         window.open(url!, '_blank');
         break;
-      case 'Destroy':
-        destroyInstance(name!, tenantDisplayName!);
+      case DropDownAction.destroy:
+        deleteInstanceMutation({
+          variables: {
+            instanceId: name,
+            tenantNamespace: tenantNamespace!,
+          },
+        });
         break;
-      case 'SSH':
+      case DropDownAction.ssh:
         setSshModal(true);
         break;
-      case 'Upload':
+      case DropDownAction.upload:
         // Something to add
         break;
+      case DropDownAction.destroy_all:
+        // to do, popconfirm not work maybe we should use a modal for the confirmation
```

Comment:
tip: using the format `// TODO: something` could be better since certain editors can look for that string and ease tracking all the todos.